### PR TITLE
Locksystem cleanup

### PR DESCRIPTION
**Pull request by Cheackraze**
## About the PR 
cleans up some obsolete code in LockSystem
**Media**

- [x] I have added screenshots/videos to this PR showcasing its changes ingame, **or** this PR does not require an ingame showcase

**Changelog**

-cleans up some obsolete code in LockSystem

### CODE REVIEW

File: Content.Server/Lock/LockSystem.cs [hunk 1]
```diff
@@ -39,7 +41,7 @@ private void OnStartup(EntityUid uid, LockComponent lockComp, ComponentStartup a
         {
             if (EntityManager.TryGetComponent(lockComp.Owner, out AppearanceComponent? appearance))
             {
-                appearance.SetData(StorageVisuals.CanLock, true);
+                _appearanceSystem.SetData(lockComp.Owner, StorageVisuals.CanLock, true, appearance);
```
Comment (metalgearsloth) [inline Content.Server/Lock/LockSystem.cs:44]:
```suggestion
                _appearanceSystem.SetData(uid, StorageVisuals.CanLock, true, appearance);
```

File: Content.Server/Lock/LockSystem.cs [hunk 2]
```diff
@@ -91,17 +93,17 @@ public bool TryLock(EntityUid uid, EntityUid user, LockComponent? lockComp = nul
             if (!HasUserAccess(uid, user, quiet: false))
                 return false;
 
-            lockComp.Owner.PopupMessage(user, Loc.GetString("lock-comp-do-lock-success", ("entityName", EntityManager.GetComponent<MetaDataComponent>(lockComp.Owner).EntityName)));
+            _sharedPopupSystem.PopupEntity(Loc.GetString("lock-comp-do-lock-success", ("entityName", EntityManager.GetComponent<MetaDataComponent>(lockComp.Owner).EntityName)), lockComp.Owner, user);
```
Comment (metalgearsloth) [inline Content.Server/Lock/LockSystem.cs:96]:
```suggestion
            _sharedPopupSystem.PopupEntity(Loc.GetString("lock-comp-do-lock-success", ("entityName", EntityManager.GetComponent<MetaDataComponent>(uid).EntityName)), uid, user);
```

File: Content.Server/Lock/LockSystem.cs [hunk 3]
```diff
@@ -91,17 +93,17 @@ public bool TryLock(EntityUid uid, EntityUid user, LockComponent? lockComp = nul
...
             lockComp.Locked = true;
 
             if(lockComp.LockSound != null)
             {
-                SoundSystem.Play(lockComp.LockSound.GetSound(), Filter.Pvs(lockComp.Owner), lockComp.Owner, AudioParams.Default.WithVolume(-5));
+                _audio.PlayPvs(_audio.GetSound(lockComp.LockSound), lockComp.Owner, AudioParams.Default.WithVolume(-5));
```
Comment (metalgearsloth) [inline Content.Server/Lock/LockSystem.cs:101]:
```suggestion
                _audio.PlayPvs(_audio.GetSound(lockComp.LockSound), uid, AudioParams.Default.WithVolume(-5));
```

File: Content.Server/Lock/LockSystem.cs [hunk 4]
```diff
@@ -195,12 +200,12 @@ private void OnEmagged(EntityUid uid, LockComponent component, GotEmaggedEvent a
             {
                 if (component.UnlockSound != null)
                 {
-                    SoundSystem.Play(component.UnlockSound.GetSound(), Filter.Pvs(component.Owner), component.Owner, AudioParams.Default.WithVolume(-5));
+                    _audio.PlayPvs(_audio.GetSound(component.UnlockSound), component.Owner, AudioParams.Default.WithVolume(-5));
```
Comment (metalgearsloth) [inline Content.Server/Lock/LockSystem.cs:203]:
```suggestion
                    _audio.PlayPvs(_audio.GetSound(component.UnlockSound), uid, AudioParams.Default.WithVolume(-5));
```

File: Content.Server/Lock/LockSystem.cs [hunk 5]
```diff
@@ -195,12 +200,12 @@ private void OnEmagged(EntityUid uid, LockComponent component, GotEmaggedEvent a
...
                 }
 
                 if (EntityManager.TryGetComponent(component.Owner, out AppearanceComponent? appearanceComp))
                 {
-                    appearanceComp.SetData(StorageVisuals.Locked, false);
+                    _appearanceSystem.SetData(component.Owner, StorageVisuals.Locked, false, appearanceComp);
```
Comment (metalgearsloth) [inline Content.Server/Lock/LockSystem.cs:208]:
```suggestion
                    _appearanceSystem.SetData(uid, StorageVisuals.Locked, false, appearanceComp);
```

File: Content.Server/Lock/LockSystem.cs [hunk 6]
```diff
@@ -195,12 +200,12 @@ private void OnEmagged(EntityUid uid, LockComponent component, GotEmaggedEvent a
             {
                 if (component.UnlockSound != null)
                 {
```
Comment (metalgearsloth) [inline Content.Server/Lock/LockSystem.cs:202]: Can you remove the == true while you're here as well?

File: Content.Server/Lock/LockSystem.cs [hunk 7]
```diff
@@ -114,18 +116,21 @@ public void Unlock(EntityUid uid, EntityUid? user, LockComponent? lockComp = nul
...
             if (lockComp.UnlockSound != null)
             {
-                SoundSystem.Play(lockComp.UnlockSound.GetSound(), Filter.Pvs(lockComp.Owner), lockComp.Owner, AudioParams.Default.WithVolume(-5));
+                _audio.PlayPvs(_audio.GetSound(lockComp.UnlockSound), lockComp.Owner, AudioParams.Default.WithVolume(-5));
             }
 
             if (EntityManager.TryGetComponent(lockComp.Owner, out AppearanceComponent? appearanceComp))
             {
-                appearanceComp.SetData(StorageVisuals.Locked, false);
+                _appearanceSystem.SetData(lockComp.Owner, StorageVisuals.Locked, false, appearanceComp);
```
Comment (metalgearsloth) [inline Content.Server/Lock/LockSystem.cs:133]:
```suggestion
                _appearanceSystem.SetData(uid, StorageVisuals.Locked, false, appearanceComp);
```

File: Content.Server/Lock/LockSystem.cs [hunk 8]
```diff
@@ -114,18 +116,21 @@ public void Unlock(EntityUid uid, EntityUid? user, LockComponent? lockComp = nul
...
+            {
+                _sharedPopupSystem.PopupEntity(Loc.GetString("lock-comp-do-unlock-success", ("entityName", EntityManager.GetComponent<MetaDataComponent>(lockComp.Owner).EntityName)), lockComp.Owner, user.Value);
+            }
+
             lockComp.Locked = false;
 
             if (lockComp.UnlockSound != null)
             {
-                SoundSystem.Play(lockComp.UnlockSound.GetSound(), Filter.Pvs(lockComp.Owner), lockComp.Owner, AudioParams.Default.WithVolume(-5));
+                _audio.PlayPvs(_audio.GetSound(lockComp.UnlockSound), lockComp.Owner, AudioParams.Default.WithVolume(-5));
```
Comment (metalgearsloth) [inline Content.Server/Lock/LockSystem.cs:128]:
```suggestion
                _audio.PlayPvs(_audio.GetSound(lockComp.UnlockSound), uid, AudioParams.Default.WithVolume(-5));
```

File: Content.Server/Lock/LockSystem.cs [hunk 9]
```diff
@@ -114,18 +116,21 @@ public void Unlock(EntityUid uid, EntityUid? user, LockComponent? lockComp = nul
             if (!Resolve(uid, ref lockComp))
                 return;
 
-            if (user is {Valid: true})
-                lockComp.Owner.PopupMessage(user.Value, Loc.GetString("lock-comp-do-unlock-success", ("entityName", EntityManager.GetComponent<MetaDataComponent>(lockComp.Owner).EntityName)));
+            if (user is { Valid: true })
+            {
+                _sharedPopupSystem.PopupEntity(Loc.GetString("lock-comp-do-unlock-success", ("entityName", EntityManager.GetComponent<MetaDataComponent>(lockComp.Owner).EntityName)), lockComp.Owner, user.Value);
```
Comment (metalgearsloth) [inline Content.Server/Lock/LockSystem.cs:121]:
```suggestion
                _sharedPopupSystem.PopupEntity(Loc.GetString("lock-comp-do-unlock-success", ("entityName", EntityManager.GetComponent<MetaDataComponent>(lockComp.Owner).EntityName)), uid, user.Value);
```

File: Content.Server/Lock/LockSystem.cs [hunk 10]
```diff
@@ -91,17 +93,17 @@ public bool TryLock(EntityUid uid, EntityUid user, LockComponent? lockComp = nul
...
             if(lockComp.LockSound != null)
             {
-                SoundSystem.Play(lockComp.LockSound.GetSound(), Filter.Pvs(lockComp.Owner), lockComp.Owner, AudioParams.Default.WithVolume(-5));
+                _audio.PlayPvs(_audio.GetSound(lockComp.LockSound), lockComp.Owner, AudioParams.Default.WithVolume(-5));
             }
 
             if (EntityManager.TryGetComponent(lockComp.Owner, out AppearanceComponent? appearanceComp))
             {
-                appearanceComp.SetData(StorageVisuals.Locked, true);
+                _appearanceSystem.SetData(lockComp.Owner, StorageVisuals.Locked, true, appearanceComp);
```
Comment (metalgearsloth) [inline Content.Server/Lock/LockSystem.cs:106]:
```suggestion
                _appearanceSystem.SetData(uid, StorageVisuals.Locked, true, appearanceComp);
```

File: Content.Server/Lock/LockSystem.cs [hunk 11]
```diff
@@ -91,17 +93,17 @@ public bool TryLock(EntityUid uid, EntityUid user, LockComponent? lockComp = nul
...
+            _sharedPopupSystem.PopupEntity(Loc.GetString("lock-comp-do-lock-success", ("entityName", EntityManager.GetComponent<MetaDataComponent>(lockComp.Owner).EntityName)), lockComp.Owner, user);
             lockComp.Locked = true;
 
             if(lockComp.LockSound != null)
             {
-                SoundSystem.Play(lockComp.LockSound.GetSound(), Filter.Pvs(lockComp.Owner), lockComp.Owner, AudioParams.Default.WithVolume(-5));
+                _audio.PlayPvs(_audio.GetSound(lockComp.LockSound), lockComp.Owner, AudioParams.Default.WithVolume(-5));
             }
 
             if (EntityManager.TryGetComponent(lockComp.Owner, out AppearanceComponent? appearanceComp))
```
Comment (metalgearsloth) [inline Content.Server/Lock/LockSystem.cs:104]:
```suggestion
            if (EntityManager.TryGetComponent(uid, out AppearanceComponent? appearanceComp))
```

**Review by metalgearsloth — CHANGES_REQUESTED** [review]

**Review by metalgearsloth — APPROVED** [review]

**Comment by EmoGarbage404** [comment]
was this intentional? it looks like you just accidentally deleted your branch?

**Comment by Cheackraze** [comment]
ye was clearing out old branches, whoops

**Pull request merged**